sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PreparePubsubWriteDoFn.java

```diff
@@ -179,6 +201,16 @@ public void process(
               null);
       reportedLineage = topic;
     }
+    if (!usesOrderingKey && !Strings.isNullOrEmpty(message.getOrderingKey())) {
+      if (!logOrderingKeyUnconfigured) {
+        LOG.warn(
+            "Encountered Pubsub message with ordering key but this sink was not configured to "
+                + "retain ordering keys, so they will be dropped. Please set #withOrderingKeys().");
+
+        logOrderingKeyUnconfigured = true;
```

Comment:
I think this might show up as racy with thread sanitizers. Maybe just mark it volatile to avoid that?

Reply:
Looks like the field is marked as an instance field so this would only be modified/observed locally on a DoFn copy or appropriately synchronized before being retrieved from a stage executor cache.
The gotcha there is that it won't trigger just once, but once for every copy of this DoFn that's created to satisfy the current stage parallelism.